src/day8/re_frame/trace/app_state.cljs

```diff
@@ -0,0 +1,54 @@
+(ns day8.re-frame.trace.app-state
+  (:require [reagent.core :as r]
+            [clojure.string :as str]
+            [devtools.formatters.core :as cljs-devtools]
+
+            [cljs.pprint :refer [pprint]]))
+
+(defn string->css [css-string]
+  (->> (str/split (get css-string "style") #";")
+       (map #(str/split % #":"))
+       (reduce (fn [acc [property value]]
+                 (assoc acc (keyword property) value)) {})))
+
+(defn str->hiccup
+  [string]
+  (cond (= string "span")   :span
+        (= string "style")  :style
+        (= string ", ")     " "
+        :else               string))
+
+(declare jsonml->hiccup)
+
+(defn data-structure
+  [jsonml]
+  (let [expanded? (r/atom true)]
+    (fn [jsonml]
+      [:span
+        {:class (str/join " " ["re-frame-trace--object"
+                               (when @expanded? "expanded")])}
+        [:span {:class "toggle"
+                :on-click #(swap! expanded? not)}
+           (if @expanded? "▼" "▶")]
+        (jsonml->hiccup (if @expanded?
+                          (cljs-devtools/body-api-call
+                            (.-object (get jsonml 1))
+                            (.-config (get jsonml 1)))
+                          (cljs-devtools/header-api-call
+                            (.-object (get jsonml 1))
+                            (.-config (get jsonml 1)))))])))
+
+(defn jsonml->hiccup
+  [jsonml]
```

Comment:
Probably good to write some comments about the purpose of this function, link back to the Chrome version, e.t.c.